docs/howto/writing-migrations.txt

```diff
@@ -184,6 +184,45 @@ the respective field according to your needs.
   migration is running. Objects created after the ``AddField`` and before
   ``RunPython`` will have their original ``uuid``’s overwritten.
 
+.. _non-atomic-migrations:
+
+Non-atomic migrations
+~~~~~~~~~~~~~~~~~~~~~
+
+On databases that support DDL transactions (SQLite and PostgreSQL), migrations
+will be run inside a transaction. For some use-cases, like performing data
+migrations on large tables, you may want to exempt a migration from being run
+in a transaction by setting the ``atomic`` class attribute to ``False``::
+
+    from django.db import migrations
+
+    class Migration(migrations.Migration):
+
+        atomic = False
+
+``RunPython`` operations will still be run inside a transaction unless
+``atomic=False`` is passed. Here's an example of a non-atomic data migration::
+
+    from django.db import migrations
+    import uuid
+
+    def gen_uuid(apps, schema_editor):
+        MyModel = apps.get_model("myapp", "MyModel")
+        for row in MyModel.objects.iterator():
+            row.uuid = uuid.uuid4()
+            row.save()
+
+    class Migration(migrations.Migration):
+
+        atomic = False
+
+        operations = [
+            migrations.RunPython(gen_uuid, atomic=False),
```

Comment:
Related to my comment above: declaring `atomic=False` here seems a bit redundant.